### PR TITLE
hash_search.cpp

### PR DESCRIPTION
**Pull request by Arctic2333**
#### Checklist

- [x] Added description of change
- [x] Added tests and example, test passes
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by bhaumikmistry** [comment]
@Arctic2333 Please add function definitions.

**Comment by Arctic2333** [comment]
in cpp?

> @Arctic2333 Please add function definitions.

Do you mean to add usage and implementation method in cpp file?

**Comment by bhaumikmistry** [comment]
@Arctic2333  yes  for instance 
```
/**
 * Sum numbers in a vector.
 *
 * @param values Container whose values are summed.
 * @return sum of `values`, or 0.0 if `values` is empty.
 */
double sum(std::vector<double> & const values) {
    ...
}
```
Also your ```main()``` is looking for user input? 
- [ ] Please add tests and assert to demonstrate that your  functions and/or search algorithm is doing what is suppose to do.

**Comment by Arctic2333** [comment]
> @Arctic2333 yes for instance
> 
> ```
> /**
>  * Sum numbers in a vector.
>  *
>  * @param values Container whose values are summed.
>  * @return sum of `values`, or 0.0 if `values` is empty.
>  */
> double sum(std::vector<double> & const values) {
>     ...
> }
> ```
> 
> Also your `main()` is looking for user input?
> 
> * [x]  Please add tests and assert to demonstrate that your  functions and/or search algorithm is doing what is suppose to do.

[x]  Please add tests and assert to demonstrate that your  functions and/or search algorithm is doing what is suppose to do.

**Comment by cclauss** [comment]
@bhaumikmistry What more is required on this one?

File: search/hash_search.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    return key % HASHMAX;
+}
+/* void create_list(int key)
+ * The same after the remainder will be added after the same hash header
+ * To avoid conflict, zipper method is used
+ * Insert elements into the linked list in the header */
+void create_list(int key) {  // Construct hash table
+    link p, n;
+    int index;
+    n = (link) malloc(sizeof(node));
```
Comment (kvedala) [inline search/hash_search.cpp:33]: This memory allocation never gets freed!!

**Pull request merged**